src/aspire/basis/dirac.py

```diff
@@ -1,5 +1,75 @@
+import logging
+import numpy as np
+
 from aspire.basis import Basis
+from aspire.utils.matrix import roll_dim, unroll_dim
+from aspire.utils.matlab_compat import m_flatten, m_reshape
+
+logger = logging.getLogger(__name__)
 
 
 class DiracBasis(Basis):
-    pass
+    """
+    Define a derived class for Dirac basis
+    """
+    def __init__(self, sz, mask=None):
+        """
+        Initialize an object for Dirac basis
+        :param sz: The shape of the vectors for which to define the basis.
+        :param mask: A boolean _mask of size sz indicating which coordinates
+            to include in the basis (default np.full(sz, True)).
+        """
+        if mask is None:
+            mask = np.full(sz, True)
+
+        super().__init__(sz)
+
+        self._mask = m_flatten(mask)
+        self.count = np.sum(mask)
+        self._sz_prod = self.nres ** self.ndim
+
+    def _build(self):
```

Comment:
Shouldn't this be called at the end of the constructor?

Reply:
The `_build` function is called at the end of constructor of base class. In this derived class  `_build` is overrode and called at the end of constructor too.  Some rearranges were made in the new commit dfb2805.